services/src/main/java/org/exoplatform/addons/gamification/service/RealizationsService.java

```diff
@@ -30,4 +36,20 @@ List<GamificationActionsHistoryDTO> getAllRealizationsByFilter(RealizationsFilte
    *           technical identifier is not found
    */
   GamificationActionsHistoryDTO updateRealizationStatus(Long gHistoryId, HistoryStatus status, String actionLabel, Long points, String domain) throws ObjectNotFoundException;
+
+  /**
+   * compute xls from all Realizations .
+   *
+   * @param A {@link List <GamificationActionsHistoryDTO>} object
+   * @param A filename String 
+   * @return A xls string
+   * @throws IOException
+   * @throws ObjectNotFoundException GamificationActionsHistory identified by
+   *           its technical identifier is not found
+   */
+  byte[] exportXls(String fileName,
```

Comment:
The file to export can be very big, thus to avoid dealing with its content in RAM, it's better to return here an `InputStream` coming from a temp file.

Reply:
how to deal with path ; 
public static [InputStream](https://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html) newInputStream([Path](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html) path,
                         [OpenOption](https://docs.oracle.com/javase/7/docs/api/java/nio/file/OpenOption.html)... options)
                                  throws [IOException](https://docs.oracle.com/javase/7/docs/api/java/io/IOException.html)
 
 I tried this but it didn't work
  Files.newInputStream(Paths.get(temp.getAbsolutePath()), StandardOpenOption.DELETE_ON_CLOSE);
  @boubaker